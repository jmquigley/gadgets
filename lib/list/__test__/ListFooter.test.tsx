'use strict';

const debug = require('debug')('ListFooter.test');

import {mount} from 'enzyme';
import * as React from 'react';
import {getDefaultListFooterProps, ListFooter} from '../index';

test('Test retrieval of ListFooter props object', () => {
	const props = getDefaultListFooterProps();

	expect(props).toBeTruthy();
});

test('Test the creation of a ListFooter control with simple title', () => {
	const ctl = mount(
		<ListFooter title="test title" />
	);

	expect(ctl).toBeTruthy();
	debug(ctl.debug());

	expect(ctl.prop('title')).toBe('test title');
	expect(ctl.prop('disabled')).toBe(false);
	expect(ctl.prop('visible')).toBe(true);
	expect(ctl.find('.ui-list-footer').length).toBe(1);
});

// TODO: test case for validating props object creator
// TODO: disabling the ListFooter item
// TODO: make the ListFooter item invisible
