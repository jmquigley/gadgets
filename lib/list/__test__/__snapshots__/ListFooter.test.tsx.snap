// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test retrieval of ListFooter props object 1`] = `
Object {
  "className": "",
  "contentEditable": false,
  "direction": "right",
  "disabled": false,
  "focus": false,
  "height": "",
  "id": "",
  "layout": 1,
  "location": "none",
  "noedit": false,
  "nohover": false,
  "noripple": false,
  "onClick": [Function],
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": Object {},
  "title": "",
  "visible": true,
  "widget": null,
  "width": "",
  "xcss": null,
}
`;

exports[`Test the creation of a ListFooter control with simple title 1`] = `
<ListFooter
  className=""
  contentEditable={false}
  direction="right"
  disabled={false}
  focus={false}
  height=""
  id=""
  layout={1}
  location="none"
  noedit={false}
  nohover={false}
  noripple={false}
  onClick={[Function]}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={Object {}}
  title="test title"
  visible={true}
  widget={null}
  width=""
  xcss={null}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#004358",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "optionHoverColor": "#fd7400",
      }
    }
  >
    <ListFooter__ListFooterView
      className="ui-list-footer"
      contentEditable={false}
      direction="right"
      disabled={false}
      focus={false}
      height=""
      id=""
      layout={1}
      location="none"
      noedit={false}
      nohover={false}
      noripple={true}
      onClick={[Function]}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      theme={Object {}}
      title="test title"
      visible={true}
      widget={null}
      width=""
      xcss={null}
    >
      <Title
        className="ui-list-footer ListFooter__ListFooterView-kRwuUp bvEGnQ"
        contentEditable={false}
        direction="right"
        disabled={false}
        focus={false}
        height=""
        id=""
        layout={1}
        location="none"
        noedit={false}
        nohover={false}
        noripple={true}
        onClick={[Function]}
        selected={false}
        sizing="normal"
        style={Object {}}
        testing={true}
        theme={Object {}}
        title="test title"
        visible={true}
        widget={null}
        width=""
        xcss={null}
      >
        <ThemeProvider
          theme={
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#004358",
              "color": "#000000",
              "headerBackgroundColor": "#004358",
              "headerForegroundColor": "#ffffff",
              "headerHoverColor": "#ffe11a",
              "hoverColor": "#c0c0c0",
              "optionHoverColor": "#fd7400",
            }
          }
        >
          <Title__TitleView
            className="ui-title-bar ui-list-footer ListFooter__ListFooterView-kRwuUp bvEGnQ"
            layout={1}
            style={Object {}}
          >
            <div
              className="ui-title-bar ui-list-footer ListFooter__ListFooterView-kRwuUp bvEGnQ Title__TitleView-ddYmk gnejTR"
              style={Object {}}
            >
              <Title__StyledWidget
                className="ui-title"
                contentEditable={false}
                direction="right"
                disabled={false}
                focus={false}
                height=""
                id=""
                layout={1}
                location="none"
                noedit={false}
                nohover={false}
                noripple={true}
                onClick={[Function]}
                selected={false}
                sizing="normal"
                style={Object {}}
                testing={true}
                text="test title"
                theme={Object {}}
                title="test title"
                visible={true}
                widget={null}
                width=""
                xcss={
                  Array [
                    "
	text-align: left;
	flex: 1;
",
                  ]
                }
              >
                <Label
                  className="ui-title Title__StyledWidget-dupbKD bMWgzk"
                  contentEditable={false}
                  direction="right"
                  disabled={false}
                  focus={false}
                  height=""
                  id=""
                  layout={1}
                  location="none"
                  noedit={false}
                  nohover={false}
                  noripple={true}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onDoubleClick={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onUpdate={[Function]}
                  selected={false}
                  sizing="normal"
                  style={Object {}}
                  testing={true}
                  text="test title"
                  theme={Object {}}
                  title="test title"
                  useedit={false}
                  visible={true}
                  widget={null}
                  width=""
                  xcss={
                    Array [
                      "
	text-align: left;
	flex: 1;
",
                    ]
                  }
                >
                  <ThemeProvider
                    theme={
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#004358",
                        "color": "#000000",
                        "headerBackgroundColor": "#004358",
                        "headerForegroundColor": "#ffffff",
                        "headerHoverColor": "#ffe11a",
                        "hoverColor": "#c0c0c0",
                        "optionHoverColor": "#fd7400",
                      }
                    }
                  >
                    <styled.span
                      className="ui-label ui-title Title__StyledWidget-dupbKD bMWgzk"
                      contentEditable={false}
                      innerRef={[Function]}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDoubleClick={[Function]}
                      onKeyDown={[Function]}
                      onKeyPress={[Function]}
                      style={Object {}}
                      suppressContentEditableWarning={true}
                    >
                      <span
                        className="ui-label ui-title Title__StyledWidget-dupbKD bMWgzk sc-bdVaJa fLbCIZ"
                        contentEditable={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onDoubleClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyPress={[Function]}
                        style={Object {}}
                        suppressContentEditableWarning={true}
                      >
                        test title
                      </span>
                    </styled.span>
                  </ThemeProvider>
                </Label>
              </Title__StyledWidget>
            </div>
          </Title__TitleView>
        </ThemeProvider>
      </Title>
    </ListFooter__ListFooterView>
  </ThemeProvider>
</ListFooter>
`;
